src/test/java/com/puppycrawl/tools/checkstyle/LocalizedMessageTest.java

```diff
@@ -0,0 +1,230 @@
+            }
+        });
+
+        final Utf8Control control = new Utf8Control();
+        final ResourceBundle bundle = control.newBundle(
+                "com.puppycrawl.tools.checkstyle.checks.coding.messages", Locale.ENGLISH,
+                "java.class", new TestUrlsClassLoader(url), true);
+
+        assertWithMessage("Bundle should not be null when stream is not null")
+            .that(bundle)
+            .isNotNull();
+        assertWithMessage("connection should not be using caches")
+                .that(urlConnection.getUseCaches())
+                .isFalse();
+        assertWithMessage("connection should be closed")
+                .that(closed.get())
+                .isTrue();
+    }
+
+    /**
+     * Tests reload of resource bundle.
+     *
+     * @noinspection resource, IOResourceOpenedButNotSafelyClosed
+     * @noinspectionreason resource - we have no need to use try with resources in testing
+     * @noinspectionreason IOResourceOpenedButNotSafelyClosed - no need to close resources in
+     *      testing
+     */
+    @Test
+    public void testBundleReloadUrlNotNullFalseReload() throws IOException {
+        final AtomicBoolean closed = new AtomicBoolean();
+
+        final InputStream inputStream = new InputStream() {
+            @Override
+            public int read() {
+                return -1;
+            }
+
+            @Override
+            public void close() {
+                closed.set(true);
+            }
+        };
+        final URLConnection urlConnection = new URLConnection(null) {
+            @Override
+            public void connect() {
+                // no code
+            }
+
+            @Override
+            public InputStream getInputStream() {
+                return inputStream;
+            }
+        };
+        final URL url = new URL("test", null, 0, "", new URLStreamHandler() {
+            @Override
+            protected URLConnection openConnection(URL u) {
+                return urlConnection;
+            }
+        });
+
+        final Utf8Control control = new Utf8Control();
+        final ResourceBundle bundle = control.newBundle(
+                "com.puppycrawl.tools.checkstyle.checks.coding.messages", Locale.ENGLISH,
+                "java.class", new TestUrlsClassLoader(url), false);
+
+        assertWithMessage("Bundle should not be null when stream is not null")
+            .that(bundle)
+            .isNotNull();
+        assertWithMessage("connection should not be using caches")
+                .that(urlConnection.getUseCaches())
+                .isTrue();
+        assertWithMessage("connection should be closed")
+                .that(closed.get())
+                .isTrue();
+    }
+
+    @Test
+    public void testBundleReloadUrlNotNullStreamNull() throws IOException {
+        final URL url = new URL("test", null, 0, "", new URLStreamHandler() {
+            @Override
+            protected URLConnection openConnection(URL u) {
```

Comment:
`u` -> `url`

Reply:
Done.

Reply:
Unresolving so it can be seen:
> [194](https://teamcity.jetbrains.com/viewLog.html?buildId=3931968&tab=Inspection&buildTypeId=Checkstyle_IdeaInspectionsPullRequest#): openConnection() Anonymous class parameter url hides variable in containing method 

Variable name was changed to `ignore` since the value isn't used.